--- twisted.orig/src/twisted/web/http.py
+++ twisted/src/twisted/web/http.py
@@ -3389,7 +3389,7 @@ class HTTPFactory(protocol.ServerFactory
         self._logDateTimeCall = self.reactor.callLater(1, self._updateLogDateTime)
 
     def buildProtocol(self, addr: IAddress) -> Protocol | None:
-        p = protocol.ServerFactory.buildProtocol(self, addr)
+        p = super(protocol.ServerFactory, self).buildProtocol(addr)
 
         # This is a bit of a hack to ensure that the HTTPChannel timeouts
         # occur on the same reactor as the one we're using here. This could
