Index: os-twisted/src/twisted/web/_auth/basic.py
===================================================================
--- os-twisted.orig/src/twisted/web/_auth/basic.py
+++ os-twisted/src/twisted/web/_auth/basic.py
@@ -11,6 +11,7 @@ HTTP BASIC authentication.
 """
 
 
+import base64
 import binascii
 
 from zope.interface import implementer
@@ -47,7 +48,9 @@ class BasicCredentialFactory:
         L{credentials.UsernamePassword} instance.
         """
         try:
-            creds = binascii.a2b_base64(response + b"===")
+            creds = binascii.a2b_base64(
+                response + b"=" * (-len(response) % 4), validate=True,
+            )
         except binascii.Error:
             raise error.LoginFailed("Invalid credentials")
 
