--- a/src/twisted/web/client.py
+++ b/src/twisted/web/client.py
@@ -403,7 +403,7 @@
         self.path = uri.originForm
 
     def buildProtocol(self, addr):
-        p = protocol.ClientFactory.buildProtocol(self, addr)
+        p = super(protocol.ClientFactory, self).buildProtocol(addr)
         p.followRedirect = self.followRedirect
         p.afterFoundGet = self.afterFoundGet
         if self.timeout:
--- a/src/twisted/web/http.py
+++ b/src/twisted/web/http.py
@@ -3107,7 +3107,7 @@
 
 
     def buildProtocol(self, addr):
-        p = protocol.ServerFactory.buildProtocol(self, addr)
+        p = super(protocol.ServerFactory, self).buildProtocol(addr)
 
         # This is a bit of a hack to ensure that the HTTPChannel timeouts
         # occur on the same reactor as the one we're using here. This could
