diff --git a/src/twisted/web/client.py b/src/twisted/web/client.py
index 7e4642e..d952cd6 100644
--- a/src/twisted/web/client.py
+++ b/src/twisted/web/client.py
@@ -404,7 +404,7 @@ class HTTPClientFactory(protocol.ClientFactory):
         self.path = uri.originForm
 
     def buildProtocol(self, addr):
-        p = protocol.ClientFactory.buildProtocol(self, addr)
+        p = super(protocol.ClientFactory, self).buildProtocol(addr)
         p.followRedirect = self.followRedirect
         p.afterFoundGet = self.afterFoundGet
         if self.timeout:
diff --git a/src/twisted/web/http.py b/src/twisted/web/http.py
index b7afa8b..94d3523 100644
--- a/src/twisted/web/http.py
+++ b/src/twisted/web/http.py
@@ -3107,7 +3107,7 @@ class HTTPFactory(protocol.ServerFactory):
 
 
     def buildProtocol(self, addr):
-        p = protocol.ServerFactory.buildProtocol(self, addr)
+        p = super(protocol.ServerFactory, self).buildProtocol(addr)
 
         # This is a bit of a hack to ensure that the HTTPChannel timeouts
         # occur on the same reactor as the one we're using here. This could
