diff --git a/src/twisted/trial/_asynctest.py b/src/twisted/trial/_asynctest.py
index c7a7090..bfbd224 100644
--- a/src/twisted/trial/_asynctest.py
+++ b/src/twisted/trial/_asynctest.py
@@ -142,8 +142,8 @@ class TestCase(SynchronousTestCase):
         d.addCallbacks(self._cbDeferTestMethod, self._ebDeferTestMethod,
                        callbackArgs=(result,),
                        errbackArgs=(result,))
-        d.addBoth(self.deferRunCleanups, result)
         d.addBoth(self.deferTearDown, result)
+        d.addBoth(self.deferRunCleanups, result)
         return d
