https://twistedmatrix.com/trac/ticket/9462#comment:16

--- a/src/twisted/protocols/tls.py
+++ b/src/twisted/protocols/tls.py
@@ -335,7 +335,9 @@
         Initiate, or reply to, the shutdown handshake of the TLS layer.
         """
         try:
-            shutdownSuccess = self._tlsConnection.shutdown()
+            shutdownSuccess = (
+                self._tlsConnection is not None and self._tlsConnection.shutdown()
+            )
         except Error:
             # Mid-handshake, a call to shutdown() can result in a
             # WantWantReadError, or rather an SSL_ERR_WANT_READ; but pyOpenSSL
