From 15f4d955b9c3a728076ee164090f5dde79f92d48 Mon Sep 17 00:00:00 2001
From: Ilya Skriblovsky <ilyaskriblovsky@gmail.com>
Date: Sat, 28 Mar 2020 00:07:12 +0300
Subject: [PATCH] Fixed: fireSystemEvent("shutdown") was called twice

---
 src/twisted/internet/asyncioreactor.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/twisted/internet/asyncioreactor.py b/src/twisted/internet/asyncioreactor.py
index 94b47e46dff..203f8f0a0d7 100644
--- a/src/twisted/internet/asyncioreactor.py
+++ b/src/twisted/internet/asyncioreactor.py
@@ -269,7 +269,9 @@ def run(self, installSignalHandlers=True):
 
     def stop(self):
         super().stop()
-        self.callLater(0, self.fireSystemEvent, "shutdown")
+        # This will cause runUntilCurrent which in its turn
+        # will call fireSystemEvent("shutdown")
+        self.callLater(0, lambda: None)
 
 
     def crash(self):
