Index: b/setup.py
===================================================================
--- a/setup.py
+++ b/setup.py
@@ -10,7 +10,7 @@
 import sys
 
 if sys.version_info < (2, 6):
-    print >>sys.stderr, "You must use at least Python 2.6 for Twisted"
+    print("You must use at least Python 2.6 for Twisted", sys.stderr)
     sys.exit(3)
 
 if os.path.exists('twisted'):
Index: b/twisted/python/dist.py
===================================================================
--- a/twisted/python/dist.py
+++ b/twisted/python/dist.py
@@ -313,8 +313,8 @@
     for specialExclusion in ['.svn', '_preamble.py', '_preamble.pyc']:
         if specialExclusion in thingies:
             thingies.remove(specialExclusion)
-    return filter(os.path.isfile,
-                  [os.path.join(scriptdir, x) for x in thingies])
+    return list(filter(os.path.isfile,
+                       [os.path.join(scriptdir, x) for x in thingies]))
 
 
 ## Helpers and distutil tweaks
