Index: b/twisted/scripts/tap2deb.py
===================================================================
--- a/twisted/scripts/tap2deb.py
+++ b/twisted/scripts/tap2deb.py
@@ -119,7 +119,7 @@
 
 [ -r /etc/default/%(debFile)s ] && . /etc/default/%(debFile)s
 
-test -x /usr/bin/twistd%(pythonVersion)s || exit 0
+test -x /usr/bin/twistd || exit 0
 test -r $file || exit 0
 test -r /usr/share/%(debFile)s/package-installed || exit 0
 
@@ -127,7 +127,7 @@
 case "$1" in
     start)
         echo -n "Starting %(debFile)s: twistd"
-        start-stop-daemon --start --quiet --exec /usr/bin/twistd%(pythonVersion)s -- \
+        start-stop-daemon --start --quiet --exec /usr/bin/twistd -- \
                           --pidfile=$pidfile \
                           --rundir=$rundir \
                           --%(twistdOption)s=$file \
@@ -167,12 +167,14 @@
 #!/bin/sh
 update-rc.d %(debFile)s defaults >/dev/null
 invoke-rc.d %(debFile)s start
+#DEBHELPER#
 ''' % vars())
 
     debianDir.child('prerm').setContent(
     '''\
 #!/bin/sh
 invoke-rc.d %(debFile)s stop
+#DEBHELPER#
 ''' % vars())
 
     debianDir.child('postrm').setContent(
@@ -181,6 +183,7 @@
 if [ "$1" = purge ]; then
         update-rc.d %(debFile)s remove >/dev/null
 fi
+#DEBHELPER#
 ''' % vars())
 
     debianDir.child('changelog').setContent(
@@ -199,12 +202,14 @@
 Section: net
 Priority: extra
 Maintainer: %(maintainer)s
-Build-Depends-Indep: debhelper
-Standards-Version: 3.5.6
+Build-Depends-Indep: debhelper, python (>= 2.6.5-7)
+Standards-Version: 3.8.4
+XS-Python-Version: current
 
 Package: %(debFile)s
 Architecture: all
-Depends: python%(pythonVersion)s-twisted
+Depends: ${python:Depends}, python-twisted-core
+XB-Python-Version: ${python:Versions}
 Description: %(description)s
  %(longDescription)s
 ''' % vars())
@@ -237,7 +242,7 @@
     '''\
 #!/usr/bin/make -f
 
-export DH_COMPAT=1
+export DH_COMPAT=5
 
 build: build-stamp
 build-stamp:
@@ -274,9 +279,9 @@
 	dh_strip
 	dh_compress
 	dh_installchangelogs
+	dh_python2
 	dh_fixperms
 	dh_installdeb
-	dh_shlibdeps
 	dh_gencontrol
 	dh_md5sums
 	dh_builddeb
